modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2ServiceAbstractIntegrationTest.java

```diff
@@ -481,4 +495,109 @@ public String requestAccessToken(String consumerKey, String consumerSecret,
+		appDTO.setApplicationName(OAUTH_APPLICATION_NAME);
+		appDTO.setCallbackUrl(callBackURL);
+		appDTO.setOAuthVersion(OAuth2Constant.OAUTH_VERSION_2);
+		appDTO.setGrantTypes("authorization_code implicit password client_credentials");
+		return appDTO;
+	}
+
+	/**
+	 * Register a service provider and setup consumer key and secret when a OAuthConsumerAppDTO is given.
+	 *
+	 * @param appDTO	OAuthConsumerAppDTO of the service provider.
+	 * @return			Registered service provider.
+	 * @throws Exception
+	 */
+	public ServiceProvider registerServiceProviderWithOAuthInboundConfigs(OAuthConsumerAppDTO appDTO)
+			throws Exception {
+
+		adminClient.registerOAuthApplicationData(appDTO);
+
+		OAuthConsumerAppDTO oauthConsumerApp = adminClient.getOAuthAppByName(appDTO.getApplicationName());
+		consumerKey = oauthConsumerApp.getOauthConsumerKey();
+		consumerSecret = oauthConsumerApp.getOauthConsumerSecret();
+
+		ServiceProvider serviceProvider = new ServiceProvider();
+		serviceProvider.setApplicationName(SERVICE_PROVIDER_NAME);
+		appMgtclient.createApplication(serviceProvider);
+
+		serviceProvider = appMgtclient.getApplication(SERVICE_PROVIDER_NAME);
+
+		List<InboundAuthenticationRequestConfig> authRequestList = new ArrayList<>();
+		setInboundOAuthConfig(authRequestList);
+
+		if (authRequestList.size() > 0) {
+			serviceProvider.getInboundAuthenticationConfig()
+					.setInboundAuthenticationRequestConfigs(authRequestList.toArray(
+							new InboundAuthenticationRequestConfig[authRequestList.size()]));
+		}
+		appMgtclient.updateApplicationData(serviceProvider);
+		return appMgtclient.getApplication(SERVICE_PROVIDER_NAME);
+	}
+
+	// Update app with inbound configurations.
+	private void setInboundOAuthConfig(List<InboundAuthenticationRequestConfig> authRequestList) {
+
+		if (consumerKey != null) {
+			InboundAuthenticationRequestConfig opicAuthenticationRequest =
+					new InboundAuthenticationRequestConfig();
+			opicAuthenticationRequest.setInboundAuthKey(consumerKey);
+			opicAuthenticationRequest.setInboundAuthType("oauth2");
+			if (consumerSecret != null && !consumerSecret.isEmpty()) {
+				Property property = new Property();
+				property.setName("oauthConsumerSecret");
+				property.setValue(consumerSecret);
+				Property[] properties = {property};
+				opicAuthenticationRequest.setProperties(properties);
+			}
+			authRequestList.add(opicAuthenticationRequest);
+		}
+	}
+
+	/**
+	 * Generate a X509 public certificate using the public key and private key.
+	 * @param publicKey		RSAPublicKey for the certificate.
+	 * @param privateKey	RSAPrivateKey for the certificate.
+	 * @return
+	 * @throws CertificateEncodingException
+	 * @throws NoSuchAlgorithmException
+	 * @throws SignatureException
+	 * @throws InvalidKeyException
+	 */
+	public X509Certificate getX509PublicCert(RSAPublicKey publicKey, RSAPrivateKey privateKey)
+			throws CertificateEncodingException, NoSuchAlgorithmException, SignatureException, InvalidKeyException {
```

Comment:
Shall we obey the 120 column limit here?

Reply:
It ends at 117. :-)